### ISSUE TITLE
SpectrumIdentificationResult and DB Search

### ISSUE THREAD

**Issue by mrForce** [MSGFPlus/msgfplus#51]
I guess this isn't an issue, so much as a question, but I'm using MSGF+ to search a database of peptides of lengths 8-11 amino acids. I've discovered that some of the matches (i.e. the SpectrumIdentificationResults in the resulting mzid file, right?) are to 7-mers, which aren't in my sequence database (either the targets, or the reversed decoys).

This is the command I'm using to run it:

`java -Xmx30000M -jar MSGFPlus.jar -s jlb20180807_SKarandikar_20180806EG7prep.mgf -d targets.fasta -e 9 -tda 1 -o search.mzid -addFeatures`

 I use the -e 9 option so that no digestion occurs (right?). Why are some of the resulting PSMs not in the sequence DB (or the decoy DB)? How can I change this?

**Comment by mrForce** [comment]
My version is precisely v2018.04.09.

**Comment by mrForce** [comment]
I'm going to try it with the 10/15/18 version (the one released a few days ago), and see if it has the same issue.

**Comment by mrForce** [comment]
Yup, I just tried it with the new version (though note that the Mzid file says that the program version is Release (v2018.09.12). I'm not sure why), and it still has the same issue. I'm attaching the Mzid file to this comment, in case you'd like to take a look. 
[search.mzid.zip](https://github.com/MSGFPlus/msgfplus/files/2493461/search.mzid.zip)

**Comment by mrForce** [comment]
Hey, any updates on this?

**Comment by FarmGeek4Life** [comment]
I haven't been able to get back to this; also, the source mgf file and fasta file will probably be needed to figure out what is going on.

**Comment by mrForce** [comment]
Okay, I started poking around with a debugger, but I haven't been able to figure out what's causing the problem. How should I send the MGF and FASTA file to you?

**Comment by FarmGeek4Life** [comment]
if they are bigger than 25MB when zipped, then sharing via Google Drive/OneDrive/DropBox/etc is a good option. If those won't work, we can set something else up.

**Comment by mrForce** [comment]
[mgf.zip](https://github.com/MSGFPlus/msgfplus/files/2616283/mgf.zip)

**Comment by mrForce** [comment]
The FASTA file (zipped up): 
[targets.fasta.zip](https://github.com/MSGFPlus/msgfplus/files/2616288/targets.fasta.zip)

**Comment by mrForce** [comment]
One of the 7-mers it found was TAFPSAL. The MSGF+ version was v2018.10.15.

**Comment by mrForce** [comment]
@FarmGeek4Life Any updates?

**Comment by mrForce** [comment]
@FarmGeek4Life Hey, do you have any updates on this? If not, I'm going to start trying to fix it myself. I found that the pepSeq variable on line 661 of this: https://github.com/MSGFPlus/msgfplus/blob/843235e9f9d80b6af60265f6a53d27939604e7fa/src/main/java/edu/ucsd/msjava/msdbsearch/DBScanner.java often gets set to a 7-mer, even when I use -e 9 with a FASTA consisting of 8-11 mers. It looks like the method it is contained in is for matching spectrum with sequences; is this correct?

**Comment by FarmGeek4Life** [comment]
Unfortunately, I am not very familiar with that portion of the MS-GF+ code. I haven't been able to look into this issue yet because of other tasks that had higher priority. @sangtaekim can give more information on that portion of the code, if he has the time to do so.

**Comment by mrForce** [comment]
Ah, okay. Thank you!

**Comment by mrForce** [comment]
I did a little more work this morning, and figured out the problem. It's not really a bug in MSGF+, but all of the 7-mers discovered appear to be the result of Methionine cleavage, which is represented by the option "ignoreMetCleavage" -- not enzyme as I had originally thought. Part of what threw me off is that the ignoreMetCleavage option isn't displayed in the options list that results from running MSGF+ with no parameters; I only discovered it by looking at the source code.
